mlir/lib/Dialect/Linalg/IR/LinalgInterfaces.cpp

```diff
@@ -1172,19 +1240,26 @@ LogicalResult mlir::linalg::detail::verifyStructuredOpInterface(Operation *op) {
 
     // Domain must be consistent.
     unsigned numLoops = linalgOp.getNumLoops();
-    if (indexingMap.getNumDims() != numLoops)
+    if (!isUserDefinedSemantic && indexingMap.getNumDims() != numLoops)
       return op->emitOpError("expected indexing_map #")
              << opOperand.getOperandNumber() << " to have " << numLoops
              << " dim(s) to match the number of loops";
 
     int64_t rank = linalgOp.getRank(&opOperand);
-    if (indexingMap.getNumResults() != rank)
+
+    if (!isUserDefinedSemantic && indexingMap.getNumResults() != rank)
```

Comment:
I dont see why this has to be guarded by the `isUserDefinedSemantic`. Even with that it should still satisfy the properties of `LinalgStructuredOps`.

Reply:
Yes it is, dropped it as well.